### PR TITLE
[FEATURE] Update `suite new --profile` to work with Rule Based Profiler

### PR DESCRIPTION
**Pull request by cdkini**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- CLI support for RBP
- Unit tests for underlying behavior
- Integration tests for CLI

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: great_expectations/cli/suite.py [hunk 1]
```diff
@@ -79,10 +79,13 @@ def suite(ctx):
...
     "--profile",
     "-p",
-    is_flag=True,
-    default=False,
-    help="""Generate a starting expectation suite automatically so you can refine it further. Assumes --interactive
-flag.
+    "profiler_name",
+    is_flag=False,
+    flag_value="",
+    default=None,
```
Comment (cdkini) [inline great_expectations/cli/suite.py:85]: This is somewhat going against Click patterns so the following approach is necessary.

We want to enable the following use-cases:

- `suite new` - No profiler involvement
- `suite new --profile` - Default to the legacy profiler
- `suite new --profile "my_profiler_name"` - Retrieve an RBP from store

`click.Option` can either be a flag (a boolean on/off switch) or an argument (which requires a value).

To ensure that we can treat this as both a boolean and traditional string arg, we need to use the following combination of values.

For the various options above, here's what happens to "profiler_name" in the arg list:

- `suite new` - Set to `None`
- `suite new --profile` - Set to `""`
- `suite new --profile "my_profiler_name"` - Set to `"my_profiler_name"`

File: great_expectations/cli/suite.py [hunk 2]
```diff
@@ -117,6 +120,9 @@ def suite_new(
     context: DataContext = ctx.obj.data_context
     usage_event_end: str = ctx.obj.usage_event_end
 
+    # Only set to true if `--profile` or `--profile <PROFILER_NAME>`
+    profile: bool = profiler_name is not None
+
```
Comment (cdkini) [inline great_expectations/cli/suite.py]: Perhaps we could leverage a single variable but I found a `profile` variable that acts as a boolean switch and `profiler_name` identifier easier to reason about.

This maintains the underlying structure noted in the various utility methods and seemed to be the least disruptive way to implement.

File: great_expectations/cli/suite.py [hunk 3]
```diff
@@ -529,6 +537,7 @@ def suite_edit(
         context=context,
         expectation_suite_name=expectation_suite,
         profile=False,
+        profiler_name=None,
```
Comment (cdkini) [inline great_expectations/cli/suite.py:552]: `suite edit` does not have the option of `--profile` but we need to adhere to the given signature. We could set a default of `profiler_name=None` but I thought I'd be a bit more explicit.

File: great_expectations/render/renderer/v3/suite_profile_notebook_renderer.py [hunk 4]
```diff
@@ -33,64 +34,52 @@ def __init__(
...
-            code=f"""\
-import datetime
+        self.add_header()
 
-import pandas as pd
+        # As v0.14.6, the notebook generated by `--profile` supports both the legacy UserConfigurableProfiler and the productionalized RuleBasedProfiler.
+        if self._profiler_name:
+            self._add_rule_based_profiler_cells()
+        else:
+            self._add_user_configurable_profiler_cells()
```
Comment (cdkini) [inline great_expectations/render/renderer/v3/suite_profile_notebook_renderer.py:56]: While much of the core logic is the same, there are some key differences with imports, setup, and the actual profiler call.

This doesn't feel the cleanest but is necessary to leverage this dual profiler support. If we move away from the legacy approach, this conditional goes away and the implementation can become a bit cleaner.

File: great_expectations/render/renderer/v3/suite_profile_notebook_renderer.py [hunk 5]
```diff
@@ -33,64 +34,52 @@ def __init__(
...
+        self._profiler_name = profiler_name
+        self._expectation_suite_name = self._validator.expectation_suite_name
 
     # noinspection PyMethodOverriding
-    def add_header(self) -> None:
-        self.add_markdown_cell(
-            markdown=f"""# Initialize a new Expectation Suite by profiling a batch of your data.
-This process helps you avoid writing lots of boilerplate when authoring suites by allowing you to select columns and other factors that you care about and letting a profiler write some candidate expectations for you to adjust.
+    def render(self) -> nbformat.NotebookNode:
+        self._notebook = nbformat.v4.new_notebook()
```
Comment (cdkini) [inline great_expectations/render/renderer/v3/suite_profile_notebook_renderer.py:48]: Much of this file is just reorganization to put the key methods at the top (`render` and `render_to_disk`).

The core logic you want to review is in this `render` method and the various calls it makes.

File: great_expectations/cli/suite.py (hunk 1 above)
Comment (Shinnnyshinshin) [reply great_expectations/cli/suite.py:85]: Thank you very much for this note, as well as the `help` message below 🙇🏼

**Review by Shinnnyshinshin — APPROVED** [review]
Thank you very much for this @cdkini 👍🏼

**Pull request merged**